components/google_sheets/sources/new-row-added/new-row-added.js

```diff
@@ -144,21 +156,36 @@ module.exports = {
     ) {
       return;
     }
-    const { files, newPageToken } = await this.google_drive.getModifiedSheets(
+    const { file, newPageToken } = await this.google_drive.getModifiedSheet(
       pageToken,
       this.drive === "myDrive" ? null : this.drive,
-      sheetIDs
+      this.sheetID
     );
-    for (const file of files) {
+    if (file) {
       let spreadsheet = await this.google_sheets.getSpreadsheet(file.id);
       for (const sheet of spreadsheet.sheets) {
+        if (
+          this.worksheetIDs.length > 0 &&
+          !this.worksheetIDs.includes(sheet.properties.sheetId.toString())
+        ) {
+          continue;
+        }
         let oldRowCount = this.db.get(
           `${spreadsheet.spreadsheetId}${sheet.properties.sheetId}`
         );
-        let rowCount = sheet.properties.gridProperties.rowCount;
-        if (oldRowCount && rowCount > oldRowCount) {
+        let rowCount = sheet.data[0].rowData ? sheet.data[0].rowData.length : 0;
+        if (rowCount > oldRowCount) {
           let diff = rowCount - oldRowCount;
-          this.$emit(sheet, this.getMeta(spreadsheet, sheet, diff));
+          let range = `${sheet.properties.title}!${
+            rowCount - (diff - 1)
+          }:${rowCount}`;
+          let newRowValues = await this.google_sheets.getSpreadsheetValues(
+            spreadsheet.spreadsheetId,
+            range
+          );
+          for (const newRow of newRowValues.values) {
+            this.$emit({ newRow, sheet }, this.getMeta(spreadsheet, sheet));
```

Comment:
I noticed the new emit format didn't contain the `range`. Is there a reason you left it out here? That could still be useful to emit - I think some consumers will need it.